core/parsigdb/memory.go

```diff
@@ -78,24 +78,8 @@ func (db *MemDB) StoreInternal(ctx context.Context, duty core.Duty, signedSet co
 
 // StoreExternal stores an externally received partially signed duty data set.
 func (db *MemDB) StoreExternal(ctx context.Context, duty core.Duty, signedSet core.ParSignedDataSet) error {
-	db.mu.Lock()
-	defer db.mu.Unlock()
```

Comment:
this lock was held while calling `or _, sub := range db.threshSubs {` below. This is bad practice since it unnecessarily increases the locking scope.

Reply:
how did you found out this?

Reply:
just reading through the code